subprojects/language-java/src/integTest/groovy/org/gradle/java/compile/incremental/GroovyJavaJointIncrementalCompilationIntegrationTest.groovy

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.java.compile.incremental
+
+import org.gradle.integtests.fixtures.CompiledLanguage
+import spock.lang.Unroll
+
+class GroovyJavaJointIncrementalCompilationIntegrationTest extends AbstractJavaGroovyIncrementalCompilationSupport {
+    CompiledLanguage language = CompiledLanguage.GROOVY
+
+    def setup() {
+        configureGroovyIncrementalCompilation()
+    }
+
+    static Map sources = [
+        'G': 'class G { }',
+        'G.changed': 'class G { int i }',
+        'J': 'class J { }',
+        'J.changed': 'class J { int i; }',
+
+        'G_J': 'class G_J extends J { }',
+
+        'J_G': 'class J_G extends G { }',
+
+        'G_G': 'class G_G extends G { }',
+        'J_G_G': 'class J_G_G extends G_G { }',
+    ]
+
+    void applyFileSet(List<String> fileSet) {
```

Comment:
Can you move the helper methods to the bottom of the file?

Reply:
Done.